File: 0771-jewels-and-stones.cpp
/*
771. Jewels and Stones

Submitted: October 18, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 8.56 MB (beats 26.93%)
*/

class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> j;
        for (char jewel : jewels) {
            j.insert(jewel);
        }
        int res = 0;
        for (char stone : stones) {
            res += j.count(stone);
        }
        return res;
    }
};

File: 0771-jewels-and-stones.py
"""
771. Jewels and Stones

Submitted: November 20, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 17.57 MB (beats 17.57%)
"""

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels = set(jewels)
        res = 0
        for stone in stones:
            if stone in jewels:
                res += 1
        return res